src/lib/core/say.c

```diff
@@ -1051,15 +1054,13 @@ say_internal(int level, bool check_level, const char *module,
 	     const char *filename, int line, const char *error,
 	     const char *format, va_list ap)
 {
-	int errsv = errno;
 	int total = log_vsay(log_default, level, check_level, module, filename,
 			     line, error, format, ap);
 	if (total > 0 &&
 	    level == S_FATAL && log_default->fd != STDERR_FILENO) {
 		ssize_t r = safe_write(STDERR_FILENO, say_buf, total);
 		(void)r;
 	}
-	errno = errsv; /* Preserve the errno. */
```

Comment:
Strictly speaking, `log->format_func`, which is called by `log_vsay`, may set `errno` (in `snprintf`) so I wouldn't move it to `safe_write`.

Reply:
`log_vsay` itself saves/restores `errno`, am I missing something?

Reply:
Ah, okay, missed that.

Reply:
Still, decided to not move `errno` saving code from `say_internal` to `safe_write`, because `safe_write` is called from `log_vsay` so this change doesn't make the code any better.